Core/Prognetics.Saga.Core.DependencyInjection/SagaServiceCollectionExtensions.cs

```diff
@@ -25,6 +29,25 @@ public static partial class SagaServiceCollectionExtensions
         return serviceCollection;
     }
 
+    public static ISagaConfiguration AddMongoDbSagaLog(
+        this ISagaConfiguration configuration,
+        Action<MongoDbSagaLogOptions> configure)
+    {
+        configuration.Services
+            .AddOptions<MongoDbSagaLogOptions>()
+            .BindConfiguration("Saga.Log")
+            .Configure(configure);
+
+        configuration.Services
+            .AddTransient(x => x.GetRequiredService<IOptions<MongoDbSagaLogOptions>>().Value)
+            .AddSingleton<IMongoClient>(x => new MongoClient(
```

Comment:
Should db context be registered as singleton? This is an anti-pattern for relational dbs.

Reply:
http://mongodb.github.io/mongo-csharp-driver/2.17/reference/driver/connecting/#re-use-1